cpp/daal/src/algorithms/dtrees/gbt/treeshap.cpp

```diff
@@ -0,0 +1,214 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "src/algorithms/dtrees/gbt/treeshap.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace gbt
+{
+namespace treeshap
+{
+
+uint8_t getRequestedAlgorithmVersion(uint8_t fallback)
+{
+    char * val = getenv("SHAP_VERSION");
+    if (val)
+    {
+        return atoi(val);
+    }
+    return fallback;
+}
+namespace internal
+{
+
+namespace v0
+{
+
+// extend our decision path with a fraction of one and zero extensions
+void extendPath(PathElement * uniquePath, size_t uniqueDepth, float zeroFraction, float oneFraction, int featureIndex)
```

Comment:
Should we use different `fp` types?

Reply:
Could be an upgrade, the reference implementations all use `float` though